flink-runtime/src/main/java/org/apache/flink/runtime/io/network/metrics/NettyShuffleMetricFactory.java

```diff
@@ -223,4 +230,26 @@ public static void registerDebloatingTaskMetrics(
         taskGroup.gauge(
                 MetricNames.ESTIMATED_TIME_TO_CONSUME_BUFFERS, new TimeToConsumeGauge(inputGates));
     }
+
+    /**
+     * This is a small hack. Instead of spawning a custom thread to monitor {@link
+     * NetworkBufferPool} usage, we are re-using {@link View#update()} method for this purpose.
+     */
+    private static class RequestedMemoryUsageMetric implements Gauge<Integer>, View {
```

Comment:
If I am right and the value can not be changed in runtime then I suppose we don't need the View or even metric at all.

Reply:
As I explained above this is not the case.